core/domain/takeout_service_test.py

```diff
@@ -0,0 +1,489 @@
+            completed_node_ids=self.COMPLETED_NODE_IDS_1).put()
+
+        # Setup for CollectionRightsModel.
+        collection_models.CollectionRightsModel(
+            id=self.COLLECTION_IDS[0],
+            owner_ids=[self.USER_ID_1],
+            editor_ids=[self.USER_ID_1],
+            voice_artist_ids=[self.USER_ID_1],
+            viewer_ids=[self.USER_ID_1],
+            community_owned=False,
+            status=constants.ACTIVITY_STATUS_PUBLIC,
+            viewable_if_private=False,
+            first_published_msec=0.0
+        ).save(
+            'cid', 'Created new collection right',
+            [{'cmd': rights_manager.CMD_CREATE_NEW}])
+
+        # Setup for GeneralSuggestionModel.
+        suggestion_models.GeneralSuggestionModel.create(
+            suggestion_models.SUGGESTION_TYPE_EDIT_STATE_CONTENT,
+            suggestion_models.TARGET_TYPE_EXPLORATION,
+            self.EXPLORATION_IDS[0], 1,
+            suggestion_models.STATUS_IN_REVIEW, self.USER_ID_1,
+            'reviewer_1', self.CHANGE_CMD, self.SCORE_CATEGORY,
+            'exploration.exp1.thread_1')
+
+        # Setup for ExplorationRightsModel.
+        exploration_models.ExplorationRightsModel(
+            id=self.EXPLORATION_IDS[0],
+            owner_ids=[self.USER_ID_1],
+            editor_ids=[self.USER_ID_1],
+            voice_artist_ids=[self.USER_ID_1],
+            viewer_ids=[self.USER_ID_1],
+            community_owned=False,
+            status=constants.ACTIVITY_STATUS_PUBLIC,
+            viewable_if_private=False,
+            first_published_msec=0.0
+        ).save(
+            'cid', 'Created new exploration right',
+            [{'cmd': rights_manager.CMD_CREATE_NEW}])
+
+        # Setup for UserSettingsModel.
+        user_models.UserSettingsModel(
+            id=self.USER_ID_1,
+            email=self.USER_1_EMAIL,
+            role=self.USER_1_ROLE,
+            username=self.GENERIC_USERNAME,
+            normalized_username=self.GENERIC_USERNAME,
+            last_agreed_to_terms=self.GENERIC_DATE,
+            last_started_state_editor_tutorial=self.GENERIC_DATE,
+            last_started_state_translation_tutorial=self.GENERIC_DATE,
+            last_logged_in=self.GENERIC_DATE,
+            last_created_an_exploration=self.GENERIC_DATE,
+            last_edited_an_exploration=self.GENERIC_DATE,
+            profile_picture_data_url=self.GENERIC_IMAGE_URL,
+            default_dashboard='learner', creator_dashboard_display_pref='card',
+            user_bio=self.GENERIC_USER_BIO,
+            subject_interests=self.GENERIC_SUBJECT_INTERESTS,
+            first_contribution_msec=1,
+            preferred_language_codes=self.GENERIC_LANGUAGE_CODES,
+            preferred_site_language_code=(self.GENERIC_LANGUAGE_CODES[0]),
+            preferred_audio_language_code=(self.GENERIC_LANGUAGE_CODES[0])
+        ).put()
+
+        # Setup for GeneralFeedbackReplyToId.
+        user_two_fake_hash_lambda_one = (
```

Comment:
Here and below: by "fake" do you mean "deterministic"?

Reply:
This was taken from the tests that Luis wrote for GeneralFeedbackReplyToId, so I ended up reusing a lot of the terminology he used. @ulloaluis?

Reply:
I swapped the "convert_to_hash" function with a deterministic lambda function since the model creates ids with some level of randomness. I called it fake since it could be thought of as "spoofing" the "convert_to_hash" function. Feel free to rename it. I think I had a reason for not using a random seed when I worked on this before, but I can't recall why(possibly because generating the id is a private method).

Also, the comment on line 257 should probably say "GeneralFeedbackEmailReplyToId."

Reply:
Got it, thanks!